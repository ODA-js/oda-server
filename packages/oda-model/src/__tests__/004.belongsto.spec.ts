import 'jest';
import {
  BelongsTo,
  belongsToDefaultMetaInfo,
  belongsToDefaultInput,
} from '../belongsto';
import { merge } from 'lodash';

describe('BelongsTo', () => {
  it('default', () => {
    debugger;
    const res = new BelongsTo({ belongsTo: 'A#' });
    expect(res.metadata.persistence).toMatchObject(
      belongsToDefaultMetaInfo.persistence,
    );
    expect(res.toObject()).toMatchObject(
      merge({ belongsTo: 'A#id' }, belongsToDefaultInput),
    );
    expect(res.toObject()).toMatchSnapshot('toObject');
  });
  it('create by default', () => {
    const res = new BelongsTo({
      entity: 'AA',
      field: 'a',
      belongsTo: 'A#id',
      required: true,
      identity: true,
      indexed: true,
    });
    expect(res.verb).toBe('BelongsTo');
    expect(res.ref).toMatchObject({ entity: 'A', field: 'id' });
    expect(res.entity).toBe('AA');
    expect(res.field).toBe('a');
    expect(res.required).toBe(true);
    expect(res.identity).toBe(true);
    expect(res.indexed).toBe(true);
    expect(res.metadata.persistence.single).toBe(true);
    expect(res.metadata.persistence.stored).toBe(true);
    expect(res.name).toBe('AABelongsToA');
  });
});
