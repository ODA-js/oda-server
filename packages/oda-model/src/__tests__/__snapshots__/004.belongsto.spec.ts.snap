// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BelongsTo default: toObject 1`] = `
Object {
  "belongsTo": "A#id",
  "entity": undefined,
  "field": undefined,
  "metadata": Object {
    "name": Object {
      "full": "BelongsTo",
      "normal": "",
      "short": "",
    },
    "persistence": Object {
      "embedded": false,
      "single": true,
      "stored": true,
    },
    "verb": "BelongsTo",
  },
  "name": "BelongsTo",
  "opposite": undefined,
}
`;
